### DISCUSSION TITLE
How to record inference time instead of solving time

### DISCUSSION THREAD

**Discussion by dxyzx0** [ds4dm/ecole#266] [Questions]
I know how to record `SolvingTime` in [here](https://doc.ecole.ai/py/en/stable/reference/rewards.html#solving-time), but I want to record the inference time instead of total solving time. 

How to accomplish this?

**Comment by AntoinePrv** [comment]
Hi @DingXiangYuanZhiXing,

What do you count in the inference time? Only the time to evaluate the user policy? If so, you can do that in plain Python, for example:

```python
import time

import ecole

env = ecole.environment.Branching()
instances = ecole.instance.SetCoverGenerator()

for _ in range(10):
    obs, action_set, reward_offset, done, info = env.reset(next(instances))
    policy_time = 0
    while not done:
        start = time.perf_counter_ns()
        action = policy(obs, action_set)
        policy_time += time.perf_counter_ns() - start
        obs, action_set, reward, done, info = env.step(action)
    
    # Do something with `policy_time`
```
